bedrock/3_EntityWorkflow/workflows.py

```diff
@@ -0,0 +1,174 @@
+import asyncio
+from datetime import timedelta
+from temporalio import workflow
+from collections import deque
+from typing import List, Tuple, Deque, Optional
+from dataclasses import dataclass
+
+with workflow.unsafe.imports_passed_through():
+    from activities import prompt_bedrock
+
+
+@dataclass
+class BedrockParams:
+    conversation_summary: Optional[str] = None
+    prompt_queue: Optional[Deque[str]] = None
+
+
+@workflow.defn
+class EntityBedrockWorkflow:
+    def __init__(self) -> None:
+        # List to store prompt history
+        self.conversation_history: List[Tuple[str, str]] = (
+            []
+        )
+        self.prompt_queue: Deque[str] = deque()
+        self.conversation_summary: Optional[str] = None
+        self.continue_as_new_per_turns: int = 6
+        self.chat_ended: bool = False
+
+    @workflow.run
+    async def run(
+        self,
+        params: BedrockParams = None,
+    ) -> str:
+
+        if params and params.conversation_summary:
+            self.conversation_history.append(
+                ("conversation_summary", params.conversation_summary)
+            )
+
+        if params and params.prompt_queue:
+            for prompt in params.prompt_queue:
+                self.prompt_queue.append(prompt)
+
+        summary_activity_task: Optional[asyncio.Task] = None
+
+        while True:
+            workflow.logger.info("Waiting for prompts...")
+
+            # Wait for a chat message (signal) or timeout
+            await workflow.wait_condition(
+                lambda: bool(self.prompt_queue) or self.chat_ended
+            )
+
+            # if end chat signal was sent
+            if self.chat_ended:
```

Comment:
This will lose prompts if they sent a prompt signal _and_ a chat end signal

Reply:
Good catch. Have flipped this so if both are received then all prompts are processed first.